applications/CoSimulationApplication/tests/test_rigid_body_solver.py

```diff
@@ -0,0 +1,2047 @@
+
+
+        for i in range(6):
+            for j in range(6):
+                self.assertAlmostEqual(simulation.LHS[i][j], LHS[i][j])
+
+
+    def test_check_variables(self):
+        simulation = RigidBodySolver(self.model, self.default_parameters)
+        # # Skip __init__
+        # simulation = object.__new__(RigidBodySolver)
+        # # Creating model
+        # simulation.model = KM.Model()
+        # simulation.main_model_part = simulation.model.CreateModelPart("Main")
+        # simulation.rigid_body_model_part = simulation.main_model_part.CreateSubModelPart("RigidBody")
+        # simulation.root_point_model_part = simulation.main_model_part.CreateSubModelPart("RootPoint")
+        # # Adding variables
+        # simulation.AddVariables()
+
+        # Kinematic variables (work with both RigidBody and RootPoint model parts)
+        self.assertTrue(simulation.main_model_part.HasNodalSolutionStepVariable(KM.DISPLACEMENT))
+        self.assertTrue(simulation.main_model_part.HasNodalSolutionStepVariable(KM.ROTATION))
+        self.assertTrue(simulation.main_model_part.HasNodalSolutionStepVariable(KM.VELOCITY))
+        self.assertTrue(simulation.main_model_part.HasNodalSolutionStepVariable(KM.ANGULAR_VELOCITY))
+        self.assertTrue(simulation.main_model_part.HasNodalSolutionStepVariable(KM.ACCELERATION))
+        self.assertTrue(simulation.main_model_part.HasNodalSolutionStepVariable(KM.ANGULAR_ACCELERATION))
+        
+        # Specific variables for the model part RigidBody
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KM.FORCE))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KM.MOMENT))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KMC.PRESCRIBED_FORCE))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KMC.PRESCRIBED_MOMENT))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KMC.EFFECTIVE_FORCE))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KMC.EFFECTIVE_MOMENT))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KM.BODY_FORCE))
+        self.assertTrue(simulation.rigid_body_model_part.HasNodalSolutionStepVariable(KM.BODY_MOMENT))
+        
+        # Specific variables for the model part RootPoint
+        self.assertTrue(simulation.root_point_model_part.HasNodalSolutionStepVariable(KM.REACTION))
+        self.assertTrue(simulation.root_point_model_part.HasNodalSolutionStepVariable(KM.REACTION_MOMENT))
+        self.assertTrue(simulation.root_point_model_part.HasNodalSolutionStepVariable(KMC.PRESCRIBED_DISPLACEMENT))
+        self.assertTrue(simulation.root_point_model_part.HasNodalSolutionStepVariable(KMC.PRESCRIBED_ROTATION))
+
+
+    def test_SetCompleteVector_rigid_body(self):
+        simulation = RigidBodySolver(self.model, self.default_parameters)
```

Comment:
since you are using this a lot (in all the tests), you can moce it to `def setUp(self)` method